tlscanary/modes/performance.py

```diff
@@ -60,80 +70,143 @@ def run(self):
-                                               prefs=self.args.prefs_test, get_info=True, get_certs=True,
-                                               progress=True, return_only_errors=False))
-
-            base_uri_sets.append(self.run_test(self.base_app, self.url_set, profile=self.base_profile,
-                                               prefs=self.args.prefs_base, get_info=True, get_certs=True,
-                                               progress=True, return_only_errors=False))
-
-        # extract connection speed from all scans
-        test_connections_all = []
-        for uri_set in test_uri_sets:
-            test_connections_all.append(self.extract_connection_speed(uri_set))
-
-        base_connections_all = []
-        for uri_set in base_uri_sets:
-            base_connections_all.append(self.extract_connection_speed(uri_set))
-
-        # collapse all scan data into one URI set
-        self.consolidate_connection_speed_info(test_uri_sets[0], test_connections_all)
-        self.consolidate_connection_speed_info(base_uri_sets[0], base_connections_all)
-
-        # the first URI set becomes our primary set
-        self.test_uri_set = test_uri_sets[0]
-        self.base_uri_set = base_uri_sets[0]
-
-        # new values to be inserted into response
-        for test_record in self.test_uri_set:
-            base_record = [d for d in self.base_uri_set if d[1] == test_record[1]][0]
-            test_response_time = float(test_record[2].response.connection_speed_average)
-            base_response_time = float(base_record[2].response.connection_speed_average)
-            test_speed_aggregate += test_response_time
-            base_speed_aggregate += base_response_time
-            pct_change = float((test_response_time - base_response_time) / base_response_time) * 100
-            test_record[2].response.connection_speed_change = int(pct_change)
-            # save the speed samples of the base record to the test record for now,
-            # in case we decide we want to include this in the report later
-            test_record[2].response.connection_speed_base_samples = base_record[2].response.connection_speed_samples
-
-        self.total_change = float((test_speed_aggregate - base_speed_aggregate) / base_speed_aggregate) * 100
-
-        for rank, host, result in self.test_uri_set:
-            log.log(result.as_dict())
+        # TODO:
+        # construct a custom run_args object that will be
+        # passed to each test mode, instead of just reusing ours
+
+        argv = self.args
+        test_build = self.args.test
+        self.temp_hosts = []
+
+        initialize = True
+        test_aggregate = 0
+        base_aggregate = 0
+
+        # Scans for both builds are interleaved,
+        # to try to reduce biased results caused by
+        # changing network conditions
+        for perf_scan in xrange(0, argv.scans):
+            logging.info("Starting performance scan %s out of %s" % (perf_scan+1, argv.scans))
```

Comment:
Please use the global `logger` instance instead of the `logging` module directly. There are several other occurrences of this over the code.